src/components/relationships-panel/RelationshipsPanel.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <v-card flat id="relationship-style">
+    <div class="form__row three-column mt-3">
+      <v-card flat rounded="sm" class="relationship_title gray-card px-4">
+        <v-row no-gutters class="align-center mt-5">
+
+          <v-col cols="4">
+            <v-checkbox id="heir-legal-rep-checkbox" class="mt-0" v-model="selectedRelationships"
+              :value="RelationshipTypes.HEIR_LEGAL_REP" :label="RelationshipTypes.HEIR_LEGAL_REP"
+              :rules="relationshipRules" />
+          </v-col>
+
+          <v-col cols="4">
+            <v-checkbox id="officer-checkbox" class="mt-0" v-model="selectedRelationships"
+              :value="RelationshipTypes.OFFICER" :label="RelationshipTypes.OFFICER" :rules="relationshipRules" />
+          </v-col>
+
+          <v-col cols="4">
+            <v-checkbox id="director-checkbox" class="mt-0" v-model="selectedRelationships"
+              :value="RelationshipTypes.DIRECTOR" :label="RelationshipTypes.DIRECTOR" :rules="relationshipRules" />
+          </v-col>
+
+          <v-col cols="4">
+            <v-checkbox id="shareholder-checkbox" class="mt-0" v-model="selectedRelationships"
+              :value="RelationshipTypes.SHAREHOLDER" :label="RelationshipTypes.SHAREHOLDER"
+              :rules="relationshipRules" />
+          </v-col>
+
+          <v-col cols="4">
+            <v-checkbox id="court-ordered-party-checkbox" class="mt-0" v-model="selectedRelationships"
+              :value="RelationshipTypes.COURT_ORDERED_PARTY" :label="RelationshipTypes.COURT_ORDERED_PARTY"
+              :rules="relationshipRules" />
+          </v-col>
+        </v-row>
+      </v-card>
+    </div>
+  </v-card>
+</template>
+<script lang="ts">
+
+import Vue from 'vue'
+import { Component, Watch, Emit, Prop } from 'vue-property-decorator'
+
+@Component({})
+export default class RelationshipsPanelShared extends Vue {
+  /** Draft restoration relationships */
+  @Prop({ default: ['Heir or Legal Representative'] }) readonly draftRelationships!: []
+
+  // Local properties
+  private selectedRelationships = []
+  private relationshipRules = []
+
+  // Relationship Types Enum for the template
+  RelationshipTypes = {
+    COURT_ORDERED_PARTY: 'Court Ordered Party',
+    DIRECTOR: 'Director',
+    HEIR_LEGAL_REP: 'Heir or Legal Representative',
+    OFFICER: 'Officer',
+    SHAREHOLDER: 'Shareholder'
+  }
+
+  /**
+   * Called when component is mounted.
+   * Automatically check all previously checked relationships (if any) when user continues a draft.
+   */
+  mounted (): void {
+    if (this.draftRelationships !== null) {
```

Comment:
This can never be null, because the default prop valud is used if the parent passes down a null or undefined value.

But it could be an empty array.

Do we require a default of "heir legal rep"? Maybe the prop default could be `[]` instead?

Reply:
In the story, you could test passing an array with one or more values.

Reply:
I kept this as is. This is because when using it in the create-ui, if limited restoration draft was continued, the restoration relationships getter (passed as a prop), is going to return null (getRestoration.relationships). Hence, in this case, when user selects full restoration, I will just populate my selected relationships array with heir legal rep. I tested this in create-ui.

Reply:
I believe that, if getRestoration.relationships returns null, this component will use the default prop value. So this.draftRelationships can never be null.

I think that a default of `[]` is more generic, in case this component is used elsewhere. The parent should set any initial values.

One more thing to look at: getRestoration.relationships should never return null. It should be initialized (in state model) as an empty array. And when resuming a draft and parsing it, you should do either of these:
`if (filing.restoration.relationships) this.setRestorationRelationships(filing.restoration.relationships)`
or
`this.setRestorationRelationships(filing.restoration.relationships || [])`

I prefer the first since it means we set a default value in only 1 place (store init).

What do you think?

Reply:
For Storybook purposes, you can have one story where you pass in one (or more) relationships to initially display. (And you should test the defaul props -- which you do already.)